eZ/Publish/Core/Search/Legacy/Content/WordIndexer/Gateway/DoctrineDatabase.php

```diff
@@ -0,0 +1,476 @@
+     */
+    public function bulkIndexContent(array $contentObjects)
+    {
+        foreach ($contentObjects as $contentObject) {
+            $this->index($contentObject);
+        }
+    }
+
+    /**
+     * Remove entire search index
+     */
+    public function purgeIndex()
+    {
+        $this->searchIndex->purge();
+    }
+
+    /**
+     * Index wordIndex
+     *
+     * Ported from the legacy code
+     * @see https://github.com/ezsystems/ezpublish-legacy/blob/master/kernel/search/plugins/ezsearchengine/ezsearchengine.php#L255
+     *
+     * @param Content $content
+     * @param $indexArray
+     * @param $wordIDArray
+     * @param int $placement
+     * @return int last placement
+     */
+    private function indexWords(Content $content, $indexArray, $wordIDArray, $placement = 0)
+    {
+        $contentId = $content->versionInfo->contentInfo->id;
+
+        // Count the total words in index text
+        // $totalWordCount = count($indexArray);
+
+        /*
+        // Needs to be rewritten
+        // Count the number of instances of each word
+        $wordCountArray = array_count_values($indexArray);
+        // Strip double words
+        $uniqueIndexArray = array_unique($indexArray);
+        // Count unique words
+        $uniqueWordCount = count($uniqueIndexArray);
+        */
+
+        $prevWordId = 0;
+        $contentTypeId = $content->versionInfo->contentInfo->contentTypeId;
+        $sectionId = $content->versionInfo->contentInfo->sectionId;
+        $published = $content->versionInfo->contentInfo->publicationDate;
+
+        for ($i = 0; $i < count($indexArray); $i++) {
+            $indexWord = $indexArray[$i]['Word'];
+            $contentFieldId = $indexArray[$i]['ContentClassAttributeID'];
+            $identifier = $indexArray[$i]['identifier'];
+            $integerValue = $indexArray[$i]['integer_value'];
+            $wordId = $wordIDArray[$indexWord];
+
+            if (isset($indexArray[$i + 1])) {
+                $nextIndexWord = $indexArray[$i + 1]['Word'];
+                $nextWordId = $wordIDArray[$nextIndexWord];
+            } else {
+                $nextWordId = 0;
+            }
+            // Calculate the relevance ranking for this word
+//            $frequency = ( $wordCountArray[$indexWord] / $totalWordCount );
+            $frequency = 0;
+            $this->searchIndex->addObjectWordLink($wordId, $contentId, $frequency, $placement, $nextWordId, $prevWordId, $contentTypeId, $contentFieldId, $published, $sectionId, $identifier, $integerValue);
+            $prevWordId = $wordId;
+            $placement++;
+        }
+
+        return $placement;
+    }
+
+    /**
+     * Get field meta data
+     *
+     * Based on the legacy code
+     *
+     * @todo move this elsewhere to get rid of switch statement
+     *
+     * @param Content\Field $field
+     * @return string
+     */
+    private function getMetaData(Content\Field $field)
```

Comment:
I'll appreciate any feedback on the method getMetaData. Right now it looks quite ugly. 
It returns searchable value of a field - which is needed for Rich Text type, Country type and maybe some more types I'm unaware of. In the [legacy code](https://github.com/ezsystems/ezpublish-legacy/blob/master/kernel/search/plugins/ezsearchengine/ezsearchengine.php#L83) I had a method on a translation object. In eZ Platform API I have only a ValueObject (Content\Field) containing field raw value, so I ported some of the legacy code which prepares metaData for those 2 fields (in other cases I assumed exact value can be returned - I may be wrong here).

My initial thought was to implement getMetaData for each field in eZ\Publish\Core\FieldType but I'm not sure where exactly to do it and how to use it with Content\Field ValueObject I have.
ping @andrerom